lib/alertmanager_api/silence_details.py

```diff
@@ -0,0 +1,265 @@
+import json
+from datetime import datetime
+
+
+class SilenceDetails:  # pylint: disable=too-few-public-methods
+    def __init__(
+        self,
+        instance_name=None,
+        start_time_dt=None,
+        end_time_dt=None,
+        author=None,
+        comment=None,
+        silence_id=None,
+        status=None,
+    ):
+        """
+        class to handle together all attributes describing a Silence event.
+        The values for the start and end times are datetime objects.
+        To convert a string like this (we assume it is UTC):
+
+            2025-01-22T12:50:00
+
+        into a datetime object, code like this can be used:
+
+            naive_datetime = datetime.strptime(datetime_string, datetime_format)
+            utc_datetime = naive_datetime.replace(tzinfo=timezone.utc)
+
+        :param instance_name: name of the server
+        :type instance_name: str
+        :param start_time_dt: start time of the silence event
+        :type start_time_dt: datetime
+        :param end_time_dt: end time of the silence event
+        :type end_time_dt: datetime
+        :param author: entity creating the silence event
+        :type author: str
+        :param comment: explanation for the silence event
+        :type comment: str
+        :param silence_id: the ID for a valid silence event
+        :type comment: str
+        :param status: the status of a valid silence event (pending, expired, active)
+        :type status: str
+        """
+        # we want to ensure start_time_dt and end_time_dt
+        # are really instances of type datetime
+        # and not just strings
+        if start_time_dt is not None and not isinstance(start_time_dt, datetime):
+            raise TypeError("start_time_dt must be of type datetime.")
+        if end_time_dt is not None and not isinstance(end_time_dt, datetime):
+            raise TypeError("end_time_dt must be of type datetime.")
```

Comment:
I think these are missing in the unit test, could we add a unit test for these two cases to check the error is raised?